ecosystem/gstreamer_plugin/gst_mtl_st30p_rx.c

```diff
@@ -0,0 +1,643 @@
+      gobject_class, PROP_RX_PORT_IP,
+      g_param_spec_string("rx-ip", "Sender node's IP", "Receiving MTL node IP address.",
+                          NULL, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property(
+      gobject_class, PROP_RX_PORT_UDP_PORT,
+      g_param_spec_uint("rx-udp-port", "Sender UDP port", "Receiving MTL node UDP port.",
+                        0, G_MAXUINT, 20000, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property(
+      gobject_class, PROP_RX_PORT_PAYLOAD_TYPE,
+      g_param_spec_uint("rx-payload-type", "ST 2110 payload type",
+                        "SMPTE ST 2110 payload type.", 0, G_MAXUINT, 112,
+                        G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property(
+      gobject_class, PROP_RX_PORT_RX_QUEUES,
+      g_param_spec_uint("rx-queues", "Number of RX queues",
+                        "Number of RX queues to initialize in DPDK backend.", 0,
+                        G_MAXUINT, 16, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property(
+      gobject_class, PROP_RX_FRAMERATE,
+      g_param_spec_uint("rx-fps", "Audio framerate", "Framerate of the video.", 0,
+                        G_MAXUINT, 0, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property(
+      gobject_class, PROP_RX_FRAMEBUFF_NUM,
+      g_param_spec_uint("rx-framebuff-num", "Number of framebuffers",
+                        "Number of framebuffers to be used for transmission.", 0,
+                        G_MAXUINT, 3, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property(
+      gobject_class, PROP_RX_CHANNEL,
+      g_param_spec_uint("rx-channel", "Audio channel", "Audio channel number.", 0,
+                        G_MAXUINT, 2, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property(
+      gobject_class, PROP_RX_SAMPLING,
+      g_param_spec_uint("rx-sampling", "Audio sampling rate", "Audio sampling rate.", 0,
+                        G_MAXUINT, 48000, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property(
+      gobject_class, PROP_RX_AUDIO_FORMAT,
+      g_param_spec_string("rx-audio-format", "Audio format", "Audio format type.", NULL,
+                          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+}
+
+static gboolean gst_mtl_st30p_rx_start(GstBaseSrc* basesrc) {
+  struct mtl_init_params mtl_init_params = {0};
+  struct st30p_rx_ops* ops_rx;
+  gint ret;
+
+  Gst_Mtl_St30p_Rx* src = GST_MTL_ST30P_RX(basesrc);
+  ops_rx = &src->ops_rx;
+
+  GST_DEBUG_OBJECT(src, "start");
+  GST_DEBUG("Media Transport Initialization start");
+
+  /* mtl is already initialzied */
+  if (src->mtl_lib_handle) {
+    GST_INFO("Mtl already initialized");
+    if (mtl_start(src->mtl_lib_handle)) {
+      GST_ERROR("Failed to start MTL");
+      return FALSE;
+    }
+    return TRUE;
```

Comment:
are we sure that session is stared in this case? It seems we need to call st30p_rx_create() instead of returning.

Reply:
Fixed